lib/castable.h

```diff
@@ -38,6 +38,12 @@ class ICastable {
         return dynamic_cast<const T &>(*this);
     }
 
+    /// Tries to convert the class to type T. Returns a nullptr if the cast fails.
```

Comment:
The comment is wrong, as is the original one -- as mentioned above, this throws `std::bad_cast` on failure.

Reply:
Oh, nice catch. Indeed!